types/big.go

```diff
@@ -0,0 +1,50 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+)
+
+// BigInt is a big.Int wrapper which marshals JSON to a string representation of the big number
+type BigInt big.Int
+
+func (i BigInt) MarshalText() ([]byte, error) {
+	return []byte((*big.Int)(&i).String()), nil
+}
+
+func (i *BigInt) UnmarshalText(data []byte) error {
+	i2, ok := new(big.Int).SetString(string(data), 0)
+	if !ok {
+		return fmt.Errorf("wrong format for bigInt")
+	}
+	*i = (BigInt)(*i2)
+	return nil
+}
+
+func (i *BigInt) GobEncode() ([]byte, error) {
+	return i.ToInt().GobEncode()
+}
+
+func (i *BigInt) GobDecode(buf []byte) error {
+	return i.ToInt().GobDecode(buf)
+}
+
+// String returns the string representation of the big number
+func (i *BigInt) String() string {
+	return fmt.Sprintf("%s", (*big.Int)(i))
+}
+
+// ToInt converts b to a big.Int.
+func (i *BigInt) ToInt() *big.Int {
+	return (*big.Int)(i)
+}
+
+// Add sum x+y
```

Comment:
I think adding this arithmetic function is a bit arbitrary, as `math/big.Int` supports many more arithmetic operations.  I would remove this method.

Reply:
I just added those which we are currently using. Its mainly Add and Mul. So to me its fine to have a simplified wrapper of the standard big.Int implementation